Dockerfile.scratch

```diff
@@ -1,33 +1,43 @@
 # Build stage
+# syntax = docker/dockerfile:1.4.2@sha256:443aab4ca21183e069e7d8b2dc68006594f40bddf1b15bbd83f5137bd93e80e2
 ARG goversion
-FROM golang:${goversion}-alpine as builder
+FROM --platform=${BUILDPLATFORM} golang:${goversion}-buster as base
```

Comment:
We need buster as we didn't find an alpine approach for the if statement at line https://github.com/spiffe/spire/pull/3607/files#diff-7c2204a5e25ad4691be78f06dffaac49123d7dd5ed16728dba5dec3b90f803d5R17

Reply:
That means we're statically linking against glibc, correct? Statically linking glibc is very problematic and leads to runtime crashes. We need to figure out a way to continue statically linking against musl.

Reply:
yeah, if you know anyone in the community who has experience here... Have been searching for a way, didn't find it yet.